web-common/src/features/dashboards/url-state/time-ranges/parser.ts

```diff
@@ -0,0 +1,42 @@
+import type { RillTime } from "@rilldata/web-common/features/dashboards/url-state/time-ranges/RillTime";
+import grammar from "./rill-time.cjs";
+import nearley from "nearley";
+
+const compiledGrammar = nearley.Grammar.fromCompiled(grammar);
+export function parseRillTime(rillTimeRange: string): RillTime {
+  const parser = new nearley.Parser(compiledGrammar);
+  parser.feed(rillTimeRange);
+  return parser.results[0] as RillTime;
```

Comment:
Do we need to handle the case if results are empty or nil ?

Reply:
There should always be a parse error. But for safety we could throw error if the results are empty.